Buttplug.Core/Messages/Messages.cs

```diff
@@ -110,15 +179,54 @@ public class DeviceAdded : ButtplugDeviceMessage, IButtplugMessageOutgoingOnly
         [JsonProperty(Required = Required.Always)]
         public string DeviceName;
 
+        [JsonProperty(Required = Required.Always, NullValueHandling = NullValueHandling.Ignore)]
+        public Dictionary<string, MessageAttributes> DeviceMessages =
+            new Dictionary<string, MessageAttributes>();
+
+        public DeviceAdded(uint aIndex, string aName,
+            Dictionary<string, MessageAttributes> aMessages)
+            : base(ButtplugConsts.SystemMsgId, aIndex)
+        {
+            DeviceName = aName;
+            DeviceMessages = aMessages;
+
+            MessageVersioningVersion = 1;
+            MessageVersioningPrevious = new DeviceAdded0().GetType();

```

Comment:
So maybe my C# is getting a bit rusty, but I'm not sure why you are creating a new type instance? If this is a type, can't you just store the returned type?